From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bjarne Koll <git@lynxplay.dev>
Date: Thu, 26 Feb 2026 10:25:56 +0000
Subject: [PATCH] Expose original player profile

When changing a player profile after the player connected, certain
plugins might be interested in their original, mojang authenticated,
usernames.
Expose the original player profile for this usecase.

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index a6ed6ad7884f7541d5711c203a5ef4f97c40384e..c1a87d207e26ac36f439b3b0e600a922506bdcaf 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -4103,4 +4103,10 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
 0     */
     net.kyori.adventure.util.TriState glowOverride(Entity entity);
     // KTP end - glow override
+    // KTP start - expose original game profile
+    /**
+     * {@return the player profile this player originally logged in with.}
+     */
+    com.destroystokyo.paper.profile.PlayerProfile getOriginalPlayerProfile();
+    // KTP end - expose original game profile
 }
